core/scripts/cli/voting/displayStatus.js

```diff
@@ -26,13 +26,16 @@ const displayVoteStatus = async (web3, voting, designatedVoting) => {
   const roundPhase = (await voting.getVotePhase()).toString();
   // TODO: #901 Can't access Voting.rounds in latest deployed Contract https://etherscan.io/address/0xfe3c4f1ec9f5df918d42ef7ed3fba81cc0086c5f#readContract
   // const roundStats = await voting.rounds(roundId);
-
   const currentTime = await voting.getCurrentTime();
   // If the user is using the two key contract, then the account is the designated voting contract's address
   const account = designatedVoting ? designatedVoting.address : await getDefaultAccount(web3);
   const filteredRequests = await filterRequests(pendingRequests, account, roundId, roundPhase, voting);
   const rewards = await getAvailableRewards(web3, voting, account);
-  const resolvedPrices = await getResolvedPrices(web3, voting, account);
+  // TODO(#901): MetaMask provider sometimes has trouble reading past events
+  let resolvedPrices;
+  if (web3.currentProvider.label !== "metamask") {
+    resolvedPrices = await getResolvedPrices(web3, voting, account);
```

Comment:
WDYT about handling this in getResolvedPrices instead of wrapping it in an if, just so the solution is closer to the problem?

Reply:
good idea

Reply:
done